metrics/src/main/java/tech/pegasys/pantheon/metrics/LabelledMetric.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright 2018 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.metrics;
+
```

Comment:
@FunctionalInterface?

Reply:
Yeah I was in two minds if I should actually label these `@FunctionalInterface` or if they might want to grow other methods in future.  I think I wound up doing the worst thing and just being inconsistent...

`LabelledMetric` is the most likely interface to not get any extra methods but it's not actually designed to be implemented as a lambda and isn't a lambda even in the Prometheus implementation where labels are natively supported.

So I've removed the `@FunctionalInterface` from `Counter` so we consistently omit it now. It doesn't really make any difference for our internal use but in terms of a public API I think we probably want to reserve the right to add methods.